highfidelity/hifi
# Edit handle pick plane improvements

- Add a debug overlays to visualize pick plane and pick plane hits
- Separate stretch and scale handles into their own functions
- Modify stretch handles to use the same pick plane as translation handles which is axis-aligned but rotated towards camera
- Modify scale handle to use camera-facing pick plane
- Overall cleanup of edit handles

Addresses:
https://highfidelity.manuscript.com/f/cases/15770
https://highfidelity.manuscript.com/f/cases/18397

Test Plan:
- Enter Interface in desktop mode
- Open Create and create or select any entity
- Verify all rotation handles work as expected and then leave the entity partially rotated on all axes
- Verify translation on the XZ plane (by clicking and dragging the entity anywhere outside of the handles) works as expected
- Verify all translation arrows work as expected
- Verify alt-dragging the entity anywhere outside of the handles will duplicate the entity
- Verify all axis-based stretch cubes work as expected and stress test these handles from many different angles
- Verify the center scale cube works as expected (rightward or upward movement scale the entity upward, leftward or downward movement scale the entity downward, and scaling down passed the minimum inverts and scales the entity upward again) and stress test this from many different angles as well
- Verify all above handles again (besides alt-dragging) in HMD mode
- Go to your local directory for this PR build and find the scripts/system/libraries/entitySelectionTool.js file
- Change the false to true on the following line on line 794 of the entitySelectionTool.js file (you may need to be open in Administrator mode to edit it):
var debugPickPlaneEnabled = false;
- Re-run Interface in either desktop or HMD mode
- Open Create and select any entity again
- Verify when using every handle that you see a debug plane overlay and orange debug spheres that show where the mouse ray hits the debug plane as you drag
- Verify the debug overlays still visible until grabbing another handle, deleting the entity, or reloading all scripts
